apps/system/lockscreen/js/lockscreen_inputpad.js

```diff
@@ -163,7 +167,19 @@
       var span = this.passcodeCode.childNodes[i];
       if (span) {
         if (this.states.passCodeEntered.length > i) {
+        var value = this.states.passCodeEntered.substr(i, this.states.passCodeEntered.length);
+        if ((this.states.passCodeEntered.length - 1 === i) && (keyCode && (keyCode !== 'b'))) {
+          (function(aspan, currentValue) {
+            aspan.textContent = currentValue;
+            setTimeout(function() {
+              aspan.textContent = '';
+              aspan.dataset.dot = true;
+            }, CODE_VISIBLE_TIMEOUT);
```

Comment:
Ah, I see now that this is in a loop, so there won't ever be more than one digit displayed.  It would be nice to have some way of not leaving the setTimeout around to fire later, but I guess it is harmless to do it this way.